lock_api/src/rwlock.rs

```diff
@@ -230,6 +235,39 @@ pub struct RwLock<R: RawRwLock, T: ?Sized> {
     data: UnsafeCell<T>,
 }
 
+// Copied and modified from serde
+#[cfg(feature = "enable_serde")]
+impl<R, T> Serialize for RwLock<R, T>
+where
+    R: RawRwLock,
+    T: Serialize + ?Sized,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        // FIXME It may be possible to use self.read() here instead, but using
+        // self.write() is the conservative choice.  If it *is* possible
+        // (meaning that race conditions are impossible, etc.), then we should
+        // use self.read() instead for performance reasons.
+        self.write().serialize(serializer)
```

Comment:
Serde seems to use `read` here. I think think we should match the existing behavior of serde.

https://docs.rs/serde/1.0.90/src/serde/ser/impls.rs.html#544-557

Reply:
Works for me!  Like I said in the [original issue](https://github.com/Amanieu/parking_lot/issues/46#issuecomment-486461368), I was unsure as to what the best thing to do was, and was hoping you'd make the decision.

Reply:
Now changed.